### PR TITLE
Convert to new architecture

### PR DESCRIPTION
**Pull request by scorphus**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
-## Configuration
-Set `GEM_ROOT` in your `config.fish`.
+In order to use `gem` you must specify the path to gem root. You can do so by
+adding the following snippet to your `.config/fish/config.fish` file before
+sourcing Oh My Fish.
 
-Example:
+    set GEM_ROOT #path
+
+For example:
```
Comment (bpinto) [inline README.md:30]: I like this, want to add it to plugin z? That would be helpful

File: gem.fish [hunk 2]
```diff
@@ -0,0 +1,6 @@
+function init --on-event init_gem
+  if test -n "$GEM_ROOT"
+    set -gx GEM_HOME $GEM_ROOT
+    _prepend_path $GEM_ROOT/bin
```
Comment (bpinto) [inline gem.fish]: There is no prepend path function nowadays :)

File: README.md (hunk 1 above)
Comment (scorphus) [inline README.md:30]: Sure!

File: gem.fish (hunk 2 above)
Comment (scorphus) [inline gem.fish]: Ouch, my bad. Thanks! Fixing it.

**Comment by bpinto** [comment]
You are nailing it man! Thanks for converting all those plugins, I know many people are being grateful for that.

File: README.md [hunk 3]
```diff
@@ -1,16 +1,38 @@
...
+
+bundler
+=======
+
 Ruby gem support for fish.
 
-## Install
-Add `gem` to your list of fish plugins in your `config.fish`.
+Install
+-------
```
Comment (bpinto) [inline README.md:15]: Isn't this readme template different to others?

Comment (scorphus) [inline README.md:15]: Let me check

Comment (scorphus) [inline README.md:15]: Different how? From which one, for example?

Comment (bpinto) [inline README.md:15]: Well apparently the look and feel is the same but the code is different on the headers. But it's all good!

**Pull request merged**